rotkehlchen/chain/substrate/manager.py

```diff
@@ -661,6 +661,9 @@ def wait_until_a_node_is_available(
     """
     try:
         with gevent.Timeout(seconds):
+            if len(substrate_manager.available_nodes_call_order) == 0:
+                # try to connect if we haven't done it yet
+                substrate_manager.attempt_connections()
```

Comment:
```suggestion
```

What are you doing here? Why this change? Please do not make crazy changes like this. I can't have to go through each line of a PR very carefully worried you won't introduce a bug.

This function is not designed for this. It's called `wait_until_a_node_is_available()`. It's not called `wait_until_a_node_is_available_but_also_try_connect_if_we_havent_done_so_yet()`

Reply:
The problem I got is: in the previous implemetation we were trying to connect to nodes on account addition (e.g. when you add a kusama account for the first time the app was trying to connect to kusama nodes). No it's gone (since the goal is to keep as few as possible remote queries on account modification).

So if not this, what would be a better place to connect to nodes? Or maybe we could keep doing it on account addition?

Reply:
> No it's gone (since the goal is to keep as few as possible remote queries on account modification).

There is your problem. It should not be gone. Nobody said you should remove the rpc connection at addition.